qiskit/visualization/state_visualization.py

```diff
@@ -1235,11 +1237,26 @@ def _state_to_latex_ket(data: List[complex], max_size: int = 12) -> str:
     Returns:
         String with LaTeX representation of the state vector
     """
+
+    
     num = int(np.log2(len(data)))
 
     def ket_name(i):
         return bin(i)[2:].zfill(num)
 
+    def convert_to_basis(ket,ket_basis):
+        if ket_basis == 'hadamard':
+            return ket.replace('0', '+').replace('1', '-')
+        elif ket_basis == 'comp':
+            return ket
+    def apply_basis_gate(data,ket_basis):
+        if ket_basis == 'hadamard':
+            # apply to H gate to data
+            return Operator(HGate()) @ data
```

Comment:
@jakelishman Can you help me out with this one? This seems to be failing. How do I apply the HGate to all the qubits represented in the data vector?

Reply:
@jakelishman - More specifically, how do I create a Kronecker product of H_Gates to create n-bit H Gate?

Reply:
Probably the simplest way to do this is to build up the evolution operation as a `QuantumCircuit`, say
```python
mapping = QuantumCircuit(n_qubits)
mapping.h(range(n_qubits))
```
and then use `Statevector.evolve` to modify the statevector, such as
```python
mapped_data = data.evolve(mapping)
```